sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java

```diff
@@ -961,6 +967,21 @@ public void onClaimFailed(PositionT position) {}
     this.stateAccessor.finalizeState();
   }
 
+  private void resetSplitState() {
+    synchronized (splitLock) {
```

Comment:
I wouldn't make this a separate method since we are removing the finally blocks.

This would also prevent it from being called at an inappropriate time.

Reply:
done